src/common/redux/reducers/api_key.ts

```diff
@@ -0,0 +1,40 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { type Reducer } from "redux";
+
+import { apiKeysActions } from "../actions";
+import { IApiKeysArray } from "../states/api_key";
+
+const initialState: IApiKeysArray = [];
+
+function apiKeysReducer_(
+    state = initialState,
+    action: apiKeysActions.set.TAction | apiKeysActions.removeKey.TAction,
+): IApiKeysArray {
+
+    switch (action.type) {
+        case apiKeysActions.set.ID:
```

Comment:
I guess that this is more an add/update that an infinite push to an array

Reply:
And in this case, remove is not needed, it's more a reset feature for the key of the API AI Provider than a total remove, api providers is known and fixed